x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -347,6 +347,21 @@ protected void initializeFunction() {
 
         if (isContinuous()) {
             changeCollector = function.buildChangeCollector(getConfig().getSyncConfig().getField());
+
+            if (changeCollector.isOptimized() == false) {
+                logger.warn(
+                    new ParameterizedMessage(
+                        "[{}] could not find any optimizations for continuous execution, "
+                            + "this transform might run slow, please check your configuration.",
+                        getJobId()
+                    )
+                );
+                auditor.warning(
```

Comment:
It would be nice to say which pivot values are causing the the optimization to be false. Each collector could respond with a `reason`.

Reply:
created #60364 to follow up on this